pkg/image/imagecontentsources/alternativesources.go

```diff
@@ -0,0 +1,173 @@
+package imagecontentsources
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	operatorv1alpha1scheme "github.com/openshift/client-go/operator/clientset/versioned/scheme"
+	operatorv1alpha1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1alpha1"
+	imagereference "github.com/openshift/library-go/pkg/image/reference"
+	"github.com/openshift/library-go/pkg/image/registryclient"
+)
+
+// ICSPSources holds ImageContentSourcePolicy variables to look up image sources
+type ICSPSources struct {
```

Comment:
I'd consider having a contructor for this struct: `NewICSPSources(file string, client peratorv1alpha1client.ImageContentSourcePolicyInterface) ImageResolver` all the remaining implementation details I'd hide behind private members of this package.

Reply:
thanks for the suggestion, done!